mixed/spark/common/src/main/java/com/netease/arctic/spark/SessionCatalogBase.java

```diff
@@ -0,0 +1,242 @@
+    return getSessionCatalog().listNamespaces(namespace);
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(String[] namespace)
+      throws NoSuchNamespaceException {
+    return getSessionCatalog().loadNamespaceMetadata(namespace);
+  }
+
+  @Override
+  public void createNamespace(String[] namespace, Map<String, String> metadata)
+      throws NamespaceAlreadyExistsException {
+    getSessionCatalog().createNamespace(namespace, metadata);
+  }
+
+  @Override
+  public void alterNamespace(String[] namespace, NamespaceChange... changes)
+      throws NoSuchNamespaceException {
+    getSessionCatalog().alterNamespace(namespace, changes);
+  }
+
+  @Override
+  public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException {
+    return getSessionCatalog().dropNamespace(namespace);
+  }
+
+  @Override
+  public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceException {
+    // delegate to the session catalog because all tables share the same namespace
+    return getSessionCatalog().listTables(namespace);
+  }
+
+  /** Check if the table should be loaded via target catalog. */
+  protected abstract boolean isManagedTable(Table table);
+
+  @Override
+  public Table loadTable(Identifier ident) throws NoSuchTableException {
+    if (isManagedSubTable(ident)) {
+      // if it's a sub table identifier, must be a managed table
+      return getTargetCatalog().loadTable(ident);
+    }
+
+    Table table = getSessionCatalog().loadTable(ident);
+    if (isManagedTable(table)) {
+      return getTargetCatalog().loadTable(ident);
+    }
+    return table;
+  }
+
+  protected boolean isManagedSubTable(Identifier ident) {
+    return false;
+  }
+
+  /** Check if the provider is managed by target catalog. */
+  protected abstract boolean isManagedProvider(String provider);
+
+  @Override
+  public Table createTable(
+      Identifier ident, StructType schema, Transform[] partitions, Map<String, String> properties)
+      throws TableAlreadyExistsException, NoSuchNamespaceException {
+    String provider = properties.get("provider");
+    if (isManagedProvider(provider)) {
+      return getTargetCatalog().createTable(ident, schema, partitions, properties);
+    } else {
+      // delegate to the session catalog
+      return getSessionCatalog().createTable(ident, schema, partitions, properties);
+    }
+  }
+
+  @Override
+  public Table alterTable(Identifier ident, TableChange... changes) throws NoSuchTableException {
+    Table table = getSessionCatalog().loadTable(ident);
+    if (isManagedTable(table)) {
+      return getTargetCatalog().alterTable(ident, changes);
+    } else {
+      return getSessionCatalog().alterTable(ident, changes);
+    }
+  }
+
+  @Override
+  public boolean dropTable(Identifier ident) {
+    // no need to check table existence to determine which catalog to use. if a table doesn't exist
```

Comment:
The comments here seem to have been broken due to automatic formatting.

Reply:
fixed.